service/matching/tasklist/task_list_manager.go

```diff
@@ -216,20 +232,29 @@ func NewManager(
 func (c *taskListManagerImpl) Start() error {
 	defer c.startWG.Done()
 
+	if !c.taskListID.IsRoot() && c.taskListKind == types.TaskListKindNormal {
+		var info *persistence.TaskListInfo
+		err := c.throttleRetry.Do(context.Background(), func() error {
+			var err error
+			info, err = c.db.GetTaskListInfo(c.taskListID.GetRoot())
+			return err
+		})
+		if err != nil {
+			var e *types.EntityNotExistsError
+			if !errors.As(err, &e) {
+				c.Stop()
```

Comment:
under what conditions would this occur? Could you add a comment if there's forseeable ones?

Reply:
This is an edge case, and only occur before we fully migrate partition config to database for a tasklist.
Currently, if a task list is configured with multiple partitions in dynamicconfig before the creation of the task list, a non-root partition can receive a request before the root partition and when it task list manager tries to read partition config from the root partition it will get this error.

For example, if in the dynamicconfig we set all task list to have 2 partitions by default, all the non-root partitions of newly created task lists will get this error.

This will not happen once we fully migrate partition config to database. Because in that case, root partition will always be created before non-root partitions.

Reply:
Yeah, I'll add a comment. Thanks for bringing this up.